tests/test_core/test_annotationdb.py

```diff
@@ -0,0 +1,633 @@
+    got = len(db.describe(bio_type=True, seq_name=True, identifier=True))
+    expected = 3
+    assert got == expected
+
+    """Correct number of Type values when all three arguments are True"""
+    got = len(db.describe(bio_type=True, seq_name=True, identifier=True)["Type"])
+    expected = 9
+    assert got == expected
+
+    """Correct number of identifier values when all three arguments are True"""
+    got = len(db.describe(bio_type=True, seq_name=True, identifier=True)["identifier"])
+    expected = 21
+    assert got == expected
+
+    """Correct number of SeqID values when all three arguments are True"""
+    got = len(db.describe(bio_type=True, seq_name=True, identifier=True)["SeqID"])
+    expected = 1
+    assert got == expected
+
+
+def test_ordered_values():
+    """Check if the dictonary of GFF records is correctly formed into a list for loading into the in-memory database"""
+    from cogent3.core.annotation_db import _ordered_values
+
+    """Check the values explicitly"""
+    parsed_gff = gff_parser(gff_path)
+    example_dict = list(parsed_gff)[0]
+    got = _ordered_values(example_dict)
+    expected = [
+        "sequence001",
+        "mine",
+        "gene",
+        189,
+        255,
+        ".",
+        "+",
+        ".",
+        {
+            "ID": "gene0",
+            "Dbxref": "ASAP:ABE-0000006",
+            "gene": "thrL",
+            "gene_synonym": "ECK0001",
+        },
+    ]
+    assert got == expected
+
+    """Check the data types of the resulting list"""
+
+    def return_type(values):
+        types = []
+        for v in values:
+            types.append(type(v))
+        return types
+
+    got = return_type(_ordered_values(example_dict))
+    expected = [str, str, str, int, int, str, str, str, dict]
+    assert got == expected
+
+
+def test_populate_from_file_genbank():
+    """Test that the GenBank database is populated with the correct
+    number of columns"""
+
+    db = GenbankAnnotationDb(genbank_path)
+
+    """test the number of rows populated, after skipping the
+    records without 'locus_tag' key"""
+    db.db.execute(""" SELECT * FROM Genbank """)
+    got = len(list(db.db.fetchall()))
+
+    with open_(genbank_path) as infile:
+        data = list(MinimalGenbankParser(infile.readlines()))
+
+    record = data[0]
+    features = record["features"]
+    expected = 0
+    for feature in features:
+        if "locus_tag" not in list(feature.keys()):
+            continue
+        expected += 1
+
+    assert got == expected
+
+
+def test_db_query_genbank():
+    """Test that the SQL query returns the correct
+    number of rows for different combinations of bio_type/identifier"""
+
+    import pytest
+
+    db = GenbankAnnotationDb(genbank_path)
+    with open_(genbank_path) as infile:
+        data = list(MinimalGenbankParser(infile.readlines()))
+    record = data[0]
+
+    """only the bio type is provided, along with start and end"""
```

Comment:
don't include literal strings in tests, just make it a comment since that doesn't take memory. Fix all such occurrences.